mirascope/base/vectorstores.py

```diff
@@ -0,0 +1,30 @@
+from abc import ABC, abstractmethod
+from typing import Any, ClassVar, Optional, Union
+
+from pydantic import BaseModel
+
+from mirascope.base.chunkers import BaseChunker
+from mirascope.base.embedders import BaseEmbedder
+from mirascope.openai.embedders import OpenAIEmbedder
+
+from .types import BaseVectorStoreParams, Document
+
+
+class BaseVectorStore(BaseModel, ABC):
+    vectorstore_api_key: Optional[str] = None
```

Comment:
why not just `api_key`?

Reply:
I figured since we were using api_key for llm providers, it could be confusing but maybe the context of it being in a vectorstore is enough clarity?

Reply:
Yeah I think context is enough